Snowflake.Data/Core/FileTransfer/StorageClient/SFGCSClient.cs

```diff
@@ -122,68 +133,37 @@ public FileHeader GetFileHeader(SFFileMetadata fileMetadata)
+                HttpResponseMessage response = task.Result;
 
-                        return new FileHeader
-                        {
-                            digest = digest[0],
-                            contentLength = Convert.ToInt64(contentLength[0])
-                        };
-                    }
-                }
-                catch (WebException ex)
+                if (response.IsSuccessStatusCode)
                 {
-                    HttpWebResponse response = (HttpWebResponse)ex.Response;
-                    if (response.StatusCode == HttpStatusCode.Unauthorized ||
-                        response.StatusCode == HttpStatusCode.Forbidden ||
-                        response.StatusCode == HttpStatusCode.NotFound)
-                    {
-                        fileMetadata.resultStatus = ResultStatus.NOT_FOUND_FILE.ToString();
-                        return new FileHeader();
-                    }
-                }
-            }
-            else
-            {
-                // Generate the file URL based on GCS location
-                string url = generateFileURL(fileMetadata.stageInfo.location, fileMetadata.destFileName);
-                try
-                {
-                    // Issue a HEAD request
-                    HttpWebRequest request = (HttpWebRequest)WebRequest.Create(url);
-                    request.Method = "HEAD";
-                    request.Headers.Add("Authorization", $"Bearer {AccessToken}");
-
-                    using (HttpWebResponse response = (HttpWebResponse)request.GetResponse())
-                    {
-                        var digest = response.Headers.GetValues(GCS_METADATA_SFC_DIGEST);
-                        var contentLength = response.Headers.GetValues("content-length");
+                    string digest = response.Headers.GetValues(GCS_METADATA_SFC_DIGEST).First();
+                    long contentLength = (long)response.Content.Headers.ContentLength;
 
-                        fileMetadata.resultStatus = ResultStatus.UPLOADED.ToString();
+                    fileMetadata.resultStatus = ResultStatus.UPLOADED.ToString();
 
-                        return new FileHeader
-                        {
-                            digest = digest[0],
-                            contentLength = Convert.ToInt64(contentLength[0])
-                        };
-                    }
+                    return new FileHeader
+                    {
+                        digest = digest,
+                        contentLength = contentLength
+                    };
                 }
-                catch (WebException ex)
+                else
                 {
-                    fileMetadata = HandleFileHeaderErr(ex, fileMetadata);
+                    fileMetadata.lastError = new HttpRequestException($"Response status code does not indicate success: {(int)response.StatusCode} ({response.ReasonPhrase})");
```

Comment:
The HttpRequestException.StatusCode property was only introduced in .NET 5 and above. So for .NET Framework, I was only able to get the StatusCode from the response message and not the exception